src/main/java/seedu/address/logic/parser/ParserUtil.java

```diff
@@ -124,11 +124,17 @@ public static Tag parseTag(String tag) throws ParseException {
     public static int parseInsurancePlan(String insurancePlanId) throws ParseException {
         requireNonNull(insurancePlanId);
         int trimmedInsurancePlanId;
+
         try {
             trimmedInsurancePlanId = Integer.parseInt(insurancePlanId.trim());
         } catch (NumberFormatException e) {
             throw new ParseException(MESSAGE_UNPARSABLE_INSURANCE_ID);
         }
+
+        if (trimmedInsurancePlanId < 0) {
```

Comment:
Added this as I realised that there was no check for negative numbers. If the insurance Id entered is negative, it would still allow it to be parsed in the original form.

Reply:
Good catch!